backends/aerospike.go

```diff
@@ -13,59 +15,111 @@ import (
 const setName = "uuid"
 const binValue = "value"
 
+type AerospikeDBClient interface {
+	NewUuidKey(namespace string, key string) (*as.Key, error)
+	Get(key *as.Key) (*as.Record, error)
+	Put(key *as.Key, value string, ttlSeconds int) error
+}
+
 type Aerospike struct {
+	client *as.Client
+}
+
+func (db Aerospike) Get(key *as.Key) (*as.Record, error) {
+	rec, err := db.client.Get(nil, key, binValue)
+	if err != nil {
+		return nil, formatAerospikeError(err, "GET")
+	}
+	return rec, nil
+}
+
+func (db Aerospike) Put(key *as.Key, value string, ttlSeconds int) error {
+	bins := as.BinMap{binValue: value}
+	policy := &as.WritePolicy{Expiration: uint32(ttlSeconds)}
+
+	err := db.client.Put(policy, key, bins)
+
+	return formatAerospikeError(err, "PUT")
+}
+
+func (db *Aerospike) NewUuidKey(namespace string, key string) (*as.Key, error) {
+	asKey, err := as.NewKey(namespace, setName, key)
+	if err != nil {
+		return nil, formatAerospikeError(err, "NEW_KEY")
+	}
+	return asKey, nil
+}
+
+type AerospikeBackend struct {
 	cfg     config.Aerospike
-	client  *as.Client
+	client  AerospikeDBClient
 	metrics *metrics.Metrics
 }
 
-func NewAerospikeBackend(cfg config.Aerospike, metrics *metrics.Metrics) *Aerospike {
+func NewAerospikeBackend(cfg config.Aerospike, metrics *metrics.Metrics) *AerospikeBackend {
+	if cfg.Host == "" {
+		log.Fatalf("Cannot connect to empty Aerospike host")
+	}
+	if cfg.Port <= 0 {
+		log.Fatalf("Cannot connect to Aerospike host at port %d", cfg.Port)
+	}
 	client, err := as.NewClient(cfg.Host, cfg.Port)
 	if err != nil {
-		log.Fatalf("Error creating Aerospike backend: %v", err)
-		panic("Aerospike failure. This shouldn't happen.")
+		log.Fatalf("Error creating Aerospike backend: %v", formatAerospikeError(err, "NewAerospikeBackend"))
+		panic("AerospikeBackend failure. This shouldn't happen.")
 	}
 	log.Infof("Connected to Aerospike at %s:%d", cfg.Host, cfg.Port)
 
-	return &Aerospike{
+	return &AerospikeBackend{
 		cfg:     cfg,
-		client:  client,
+		client:  &Aerospike{client},
 		metrics: metrics,
 	}
 }
 
-func (a *Aerospike) Get(ctx context.Context, key string) (string, error) {
-	asKey, err := as.NewKey(a.cfg.Namespace, setName, key)
+func (a *AerospikeBackend) Get(ctx context.Context, key string) (string, error) {
+	asKey, err := a.client.NewUuidKey(a.cfg.Namespace, key)
 	if err != nil {
 		return "", err
 	}
-	rec, err := a.client.Get(nil, asKey, "value")
+	rec, err := a.client.Get(asKey)
 	if err != nil {
 		return "", err
 	}
 	if rec == nil {
```

Comment:
According to the `aerospike-client-go` repo `README.md`: 
```
It used to return no error, but a nil Record.Bins. Now it returns ErrKeyNotFound error.

This is a significant changes, and you should search your code for all instances of Bins == nil and adapt the code accordingly.
```
The thing is, I included the check anyways because it doesn't specifically says it won't return a nil `Record` under every single scenario. Given that the overhead from a `null` check is inexpensive, I defensively included it.